formats/json/commonMain/src/kotlinx/serialization/json/internal/StreamingJsonDecoder.kt

```diff
@@ -193,7 +243,7 @@ internal open class StreamingJsonDecoder(
     }
 
     private fun handleUnknown(key: String): Boolean {
-        if (configuration.ignoreUnknownKeys) {
+        if (configuration.ignoreUnknownKeys || (discriminatorHolder?.trySkip(key) ?: false)) {
```

Comment:
Wouldn't it be better to make `trySkip` an extension function on `DiscriminatorHolder?` instead of `?: false`?

Reply:
Good point